src/services/anchor-service.ts

```diff
@@ -190,7 +190,19 @@ export default class AnchorService {
    */
   async _buildMerkleTree(candidates: Candidate[]): Promise<MerkleTree<Candidate>> {
     try {
-      const merkleTree = new MerkleTree<Candidate>(this.ipfsMerge, this.ipfsCompare);
+      if (config.merkleDepthLimit) {
+        const nodeLimit = Math.pow(2, config.merkleDepthLimit)
+        if (candidates.length > nodeLimit) {
+          logger.Warn('Found ' + candidates.length + ' valid candidates to anchor, but our '
+            + 'configured merkle tree depth limit of ' + config.merkleDepthLimit
+            + ' only permits ' + nodeLimit + ' nodes in a single merkle tree anchor proof. '
+            + 'Anchoring the first ' + nodeLimit + ' candidates and leaving the rest for a future anchor batch');
+          candidates = candidates.slice(0, nodeLimit)
+          // TODO: Should we trigger our next anchor faster than the normal interval when there
```

Comment:
This change takes the approach of just leaving extra requests around to be picked up by the next time we perform an anchor, but doesn't change our anchor schedule in any way.  We may want to consider triggering anchors more frequently if we keep having more requests than fit into a single tree, but I left that as a future improvement for now

Reply:
Yeah, I think that is likely the right approach. Anchor as soon as it gets full. We obviously have to watch out for spam though.

Reply:
Sorry, I can't tell if you're agreeing with the way I did it, or the way I describe we could do it :stuck_out_tongue: Do you want me to add a change to anchor early if the number of pending requests gets too large?

Reply:
Sorry, I think I was replying to the code comment directly. I do think triggering the anchor as soon as it get's full makes sense. Is that a large change+

Reply:
It's probably fine if we don't do anything to change the periodic anchor schedule.  If we want to push the periodic anchor back if we did a proactive anchor early, then the change is larger.

Reply:
I take it back, it's not too hard to push back the scheduled anchor too.  Will have a new patch up that does that later today

Reply:
Okay, new version up